src/TestProblem/ELBDM/DiskHeating/Init_TestProb_ELBDM_DiskHeating.cpp

```diff
@@ -0,0 +1,1208 @@
+            }
+            memset(buff, '\0', STR_SIZE);
+         }
+         fclose(input_table);
+      }
+      MPI_Bcast(DispTable_r, DispNbin, MPI_DOUBLE, 0, MPI_COMM_WORLD);
+      MPI_Bcast(DispTable_d, DispNbin, MPI_DOUBLE, 0, MPI_COMM_WORLD);
+
+      if ( MPI_Rank == 0 )
+      {
+         Aux_Message( stdout, "%s ...\n", __FUNCTION__ );
+
+         const double ParM = Disk_Mass / NPar_AllRank;
+         double Ran, RanR, RanV, R, Rold, f, f_, phi, RanVec[3];
+         Aux_Message(stdout, " Particle Mass = %13.7e\n", ParM) ;
+
+//       initialize the RNG
+         RNG = new RandomNumber_t( 1 );
+         RNG->SetSeed( 0, 1002 );
+
+         for ( long p = 0; p < NPar_AllRank; p++)
+         {
+            Time_AllRank[p] = Time[0];
+//          mass
+            Mass_AllRank[p] = ParM;
+//          label
+            Type_AllRank[p] = 3;
+            Label_AllRank[p] = 3;
+
+//          position
+            Ran  = RNG->GetValue( 0, 0.0, 1.0);
+            R = 1.0;
+            do
+            {
+               f = (1 + R) * exp(-R) + Ran - 1;
+               f_ = -R * exp(-R);
+               Rold = R;
+               R = R - f / f_;
+            }
+            while(fabs(R - Rold) / R > 1e-7);
+            RanR = Disk_R*R;
+            phi = 2*pi*RNG->GetValue( 0, 0.0, 1.0 );
+            RanVec[0] = RanR*cos(phi);
+            RanVec[1] = RanR*sin(phi);
+            RanVec[2] = 0;
+            for (int d = 0; d < 3; d++) Pos_AllRank[d][p] = RanVec[d] + Cen[d];
+
+//          velocity
+            double V_acc, sigma, r;
+            double RanV[3];
+            V_acc = sqrt(Ran*Disk_Mass/RanR);
+            Vel_AllRank[0][p] = - V_acc*sin(phi);
+            Vel_AllRank[1][p] =   V_acc*cos(phi);
+            Vel_AllRank[2][p] =   0;
+
+         } // for ( long p = 0; p < NPar_AllRank; p++)
+
+      } //if ( MPI_Rank == 0 )
+   } //if ( !AddParWhenRestartByfile )
+
+// add particles here
+   Par_AddParticleAfterInit( NNewPar, NewParAtt );
+// free memory
+   for (int v=0; v<PAR_NATT_TOTAL; v++)   delete [] NewParAtt[v];
+
+// initialize the k-space Green's function for the isolated BC.
+   if ( OPT__BC_POT == BC_POT_ISOLATED )  Init_GreenFuncK();
+
+#  if ( defined PARTICLE  &&  defined LOAD_BALANCE )
+   const double Par_Weight    = amr->LB->Par_Weight;
+#  else
+   const double Par_Weight    = 0.0;
+#  endif
+#  ifdef LOAD_BALANCE
+   const UseLBFunc_t UseLB    = USELB_YES;
+#  else
+   const UseLBFunc_t UseLB    = USELB_NO;
+#  endif
+
+   for (int lv=0; lv<MAX_LEVEL; lv++)
+   {
+      if ( MPI_Rank == 0 )    Aux_Message( stdout, "   Refining level %d ...\n", lv );
+
+      Flag_Real( lv, UseLB );
+
+      Refine( lv, UseLB );
+
+#     ifdef LOAD_BALANCE
+      LB_Init_LoadBalance( true, true, Par_Weight, true, false, lv+1 );
+#     endif
+
+      if ( MPI_Rank == 0 )    Aux_Message( stdout, "   Refining level %d ... done\n", lv );
+    } // for (int lv=OPT__UM_IC_LEVEL; lv<MAX_LEVEL; lv++)
+
+    if ( !AddParWhenRestartByfile )
```

Comment:
Clarify why the operations below only apply when disabling `AddParWhenRestartByfile`